### PR TITLE
[ML] add deployment_stats to trained model stats

### PR DESCRIPTION
**Pull request by benwtrent**
This commit adds a new field `deployment_stats` that is optionally set for models that are deployed. 

If a model does not have a deployment, it will be `null`.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetDeploymentStatsAction.java [hunk 1]
```diff
@@ -115,331 +106,6 @@ public int hashCode() {
 
         public static final ParseField DEPLOYMENT_STATS = new ParseField("deployment_stats");
 
-        public static class AllocationStats implements ToXContentObject, Writeable {
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetDeploymentStatsAction.java:118]: This is is used by two transport actions now, I opted to move it to its own class. It is pretty big regardless and probably needed its own class.

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@davidkyle I removed the get deployment stats API, adding some handling of rest versioning for model stats, and updated the deployment action to not throw on missing stuff. 

Let me know what you think.

**Review by sethmlarson — APPROVED** [review]
LGTM

**Comment by davidkyle** [comment]
> Let me know what you think.

Saddened to lose the deployment stats API I had grown accustomed to it.

Still LGTM

@elastic/ml-ui this PR changes the format of the GET model stats response

**Pull request merged**